ansible_collections/arista/avd/plugins/plugin_utils/eos_validate_state_utils/get_anta_results.py

```diff
@@ -0,0 +1,174 @@
+# Copyright (c) 2023 Arista Networks, Inc.
+# Use of this source code is governed by the Apache License 2.0
+# that can be found in the LICENSE file.
+from __future__ import annotations
+
+import logging
+from asyncio import run
+from json import loads
+
+from ansible.plugins.connection import ConnectionBase
+
+from ansible_collections.arista.avd.plugins.plugin_utils.errors import AristaAvdError
+from ansible_collections.arista.avd.roles.eos_validate_state.python_modules.constants import ACRONYM_CATEGORIES, AVD_TEST_CLASSES
+
+from .ansible_eos_device import AnsibleEOSDevice
+from .catalog import Catalog
+
+try:
+    from anta.inventory import AntaInventory
+    from anta.loader import parse_catalog, setup_logging
+    from anta.result_manager import ResultManager, TestResult
+    from anta.runner import main as anta_runner
+
+    HAS_ANTA = True
+except ImportError:
+    HAS_ANTA = False
+
+try:
+    import yaml
+
+    HAS_YAML = True
+except ImportError:
+    HAS_YAML = False
+
+
+# https://ttl255.com/yaml-anchors-and-aliases-and-how-to-disable-them/
+class NoAliasDumper(yaml.SafeDumper):
+    def ignore_aliases(self, data):
+        return True
```

Comment:
Should we move this to utils so we can use it in more places?
We aware that you may need more tricks to dump the data, if it comes from Ansible input vars, since Ansible wraps in their own unicode classes. Maybe consider taking the dumper as a kwarg as well, so you can import AnsibleDumper instead in the action plugin and use that here.

Reply:
So for now even with Ansible we don't need the AnsibleDumper from what I have seen but it is quite possible that we will need it so I have moved the NoAliasDumper to utils.

I have taken the opportunity to move the dumping logic into the Catalog object which makes much more sense so thanks for this comment!